plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/models/DeploymentResourceMapper.java

```diff
@@ -129,24 +124,12 @@ private void load(String kind) {
 
 	private void buildDeployments() {
 		if (state.compareAndSet(State.UNINITIALIZED, State.LOADING)) {
-			for (String kind : WatchManager.KINDS) {
-				load(kind);
-			}
 			WatchManager.getInstance().stopWatch(project);
+			ConnectionsRegistrySingleton.getInstance().removeListener(connectionListener);
```

Comment:
We may not need that.  Part of a watching a resource involves retrieving the version from which to begin a watch and you use that when starting it; you then get all the changes since that version.  It would probably make more sense to try and capture the resource version from our initial load and start from there.  That would mean we might get duplicates but then at least we have a way to filter them out.  Additionally, projects are intended to be small.  We should hope users will not have large, copious amounts of resources.